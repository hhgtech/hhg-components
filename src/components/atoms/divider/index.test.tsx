import React from 'react'

import { render, cleanup } from '@testing-library/react'
import 'jest-styled-components'

import { Divider } from '.'
import MockTheme from '../../../misc/mockTheme'

afterEach(cleanup)

test('it renders', () => {
  const alert = render(
    <MockTheme>
      <Divider type="dotted" />
    </MockTheme>,
  )

  const self = alert.container.firstChild

  expect(self).toMatchSnapshot()
  expect(self).toHaveAttribute('data-diver-type', 'dotted')
})
