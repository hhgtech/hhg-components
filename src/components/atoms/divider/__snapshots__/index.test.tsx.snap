// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`it renders 1`] = `
.c0 {
  border-color: #d2d6dc;
  border-style: solid;
  border-width: 1px;
}

.c0[data-diver-type='dashed'] {
  border-style: dashed;
}

.c0[data-diver-type='dotted'] {
  border-style: dotted;
}

.c0[data-diver-type='double'] {
  border-style: double;
}

.c0[data-diver-type='groove'] {
  border-style: groove;
}

.c0[data-diver-type='ridge'] {
  border-style: ridge;
}

.c0[data-diver-type='inset'] {
  border-style: inset;
}

.c0[data-diver-type='outset'] {
  border-style: outset;
}

<hr
  class="c0"
  data-diver-type="dotted"
/>
`;
